WireAuthentication/Sources/WireAuthenticationUI/Views/Registration/VerificationEmailCode/VerificationEmailCodeViewModel.swift

```diff
@@ -0,0 +1,213 @@
+                password: password,
+                verificationCode: nil
+            )
+            let authenticationResult = try await createAuthenticationResult(
+                cookies: cookies,
+                emailCredentials: emailCredentials,
+                userID: uuid
+            )
+            onFlowCompletion(authenticationResult)
+        } catch {
+            WireLogger.authentication.error("register personal account failed: \(error)")
+
+            switch error {
+            case RegisterPersonalAccountUseCaseError.invalidEmail:
+                alert = .invalidEmailForRegistration
+            case RegisterPersonalAccountUseCaseError.blacklistedEmail:
+                alert = .blacklistedEmail
+            case RegisterPersonalAccountUseCaseError.tooManyTeamMembers:
+                alert = .tooManyTeamMembers
+            case RegisterPersonalAccountUseCaseError.userCreationRestricted:
+                alert = .userCreationRestricted
+            case RegisterPersonalAccountUseCaseError.invalidCode:
+                alert = .invalidCode
+            case RegisterPersonalAccountUseCaseError.emailExists:
+                alert = .emailExists
+            default:
+                router.presentAlert(for: error)
+            }
+        }
+
+        isLoading = false
+
+    }
+
+    func requestVerificationCode() async {
+        isResending = true
+
+        do {
+            try await resendVerificationCode(email: email)
+            WireLogger.authentication.info("Resend email erification code succeeded")
+        } catch {
+            WireLogger.authentication.error("Resend email erification code login failed: \(error)")
+
+            switch error {
+            case RequestEmailVerificationCodeUseCaseFailure.invalidEmail:
+                alert = .invalidEmailForRegistration
+            case RequestEmailVerificationCodeUseCaseFailure.blacklistedEmail:
+                alert = .blacklistedEmail
+            case RequestEmailVerificationCodeUseCaseFailure.emailExists:
+                alert = .emailExists
+            case RequestEmailVerificationCodeUseCaseFailure.domainBlockedForRegistration:
+                alert = .domainBlockedForRegistration
+            default:
+                router.presentAlert(for: error)
+            }
+        }
+
+        isResending = false
+    }
+
+    // MARK: - Private
+
+    private func resendVerificationCode(email: String) async throws {
+        let useCase = try await factory.requestEmailVerificationCodeUseCase()
+        try await Task.detached {
+            try await useCase.invoke(email: email)
+        }.value
+    }
+
+    private func createAuthenticationResult(
+        cookies: [HTTPCookie],
+        emailCredentials: EmailCredentials,
+        userID: UUID
+    ) async throws -> AuthenticationResult {
+        let useCase = factory.createAuthenticationResultUseCase()
+        return try await Task.detached {
+            try await useCase.invoke(
```

Comment:
question: I don't understand the reason behind creating a detached task. Why does it have to be like this?

Reply:
In this case, the `Task.detached` isn't strictly necessary. It can likely be removed for simplicity.